crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -281,6 +281,42 @@ def prepare(options = {})
 
       protected
 
+      def upgrade_first_cluster_node(node_name, next_node_name)
+        save_upgrade_state("Starting the upgrade of node #{node_name}")
+        node_api = Api::Node.new node_name
+        next_node_api = Api::Node.new next_node_name
+        # upgrade the first node
+        return false unless node_api.upgrade
+
+        # Explicitly mark the first node as cluster founder
+        # and in case of DRBD setup, adapt DRBD config accordingly.
+        unless Api::Pacemaker.set_node_as_founder node_name
+          save_error_state("Changing the cluster founder to #{node_name} has failed")
+          return false
+        end
+        # remove pre-upgrade attribute, so the services can start
+        return false unless next_node_api.disable_pre_upgrade_attribute_for node_name
+        # delete old pacemaker resources (from the node where old pacemaker is running)
+        return false unless delete_pacemaker_resources next_node_name
+        # start crowbar-join at the first node
+        return false unless node_api.post_upgrade
+        return false unless node_api.join_and_chef
+        # migrate routers from nodes being upgraded
+        node_api.router_migration
+      end
+
+      def upgrade_next_cluster_node(node_name)
```

Comment:
It's a bit misdirecting in a different way. True, once the first node is upgraded, it has to change itself to be the founder (even if it originally wasn't), but that's exactly it, it might not have been founder at the time we call this method.

Actually here "next" sounds still correct to me - we are really calling this for every other node that is not the first one.